src/analysis/convex/basic.lean

```diff
@@ -279,6 +279,18 @@ lemma convex.add_smul_mem (h : convex s) {x y : E} (hx : x ∈ s) (hy : x + y 
   {t : ℝ} (ht : t ∈ Icc (0 : ℝ) 1) : x + t • y ∈ s :=
 by { convert h.add_smul_sub_mem hx hy ht, abel }
 
+lemma convex.smul_mem_of_zero_mem (h : convex s) {x : E} (zero_mem : (0:E) ∈ s) (hx : x ∈ s)
+  {t : ℝ} (ht : t ∈ Icc (0 : ℝ) 1) : t • x ∈ s :=
+by simpa using h.add_smul_mem zero_mem (by simpa using hx) ht
+
+lemma convex.mem_smul_of_zero_mem (h : convex s) {x : E} (zero_mem : (0:E) ∈ s) (hx : x ∈ s)
+  {t : ℝ} (ht : t ∈ Ici (1 : ℝ)) : x ∈ t • s :=
+begin
+  rw mem_smul_set_iff_inv_smul_mem,
+  apply h.smul_mem_of_zero_mem zero_mem hx ⟨inv_nonneg.2 (zero_le_one.trans ht), inv_le_one ht⟩,
+  exact ne_of_gt (zero_lt_one.trans_le ht),
```

Comment:
```suggestion
  rw mem_smul_set_iff_inv_smul_mem (zero_lt_one.trans_le ht).ne,
  exact h.smul_mem_of_zero_mem zero_mem hx ⟨inv_nonneg.2 (zero_le_one.trans ht), inv_le_one ht⟩,
```
I don't know how to suggest stuff on several lines. Also I'm in the orange bar hell, but this should work.

Reply:
As I said to Eric earlier, this doesn't work.

Reply:
Whoops. And what about `rw mem_smul_set_iff_inv_smul_mem (ne_of_gt (zero_lt_one.trans_le ht))`? I still can't test it.